### PR TITLE
HTTP: Allow disabling SNI (and other things) per connection

### PR DESCRIPTION
**Pull request by ktoso**
This is an ugly resolution for #20214, will clean-up soon.

### CODE REVIEW

File: akka-http-tests/src/test/scala/akka/http/scaladsl/server/TestServer.scala [hunk 1]
```diff
@@ -73,3 +73,44 @@ object TestServer extends App {
...
+
+import akka.actor._
+import akka.http.scaladsl._
+import model._
+import akka.stream._
+import scaladsl._
+
+import com.typesafe.sslconfig.akka.AkkaSSLConfig
+
+object AkkaSni extends App {
```
Comment (ktoso) [inline akka-http-tests/src/test/scala/akka/http/scaladsl/server/TestServer.scala]: TODO: remove me

File: akka-http-core/src/main/scala/akka/http/scaladsl/ConnectionContext.scala [hunk 2]
```diff
@@ -22,11 +23,12 @@ object ConnectionContext {
   //#https-context-creation
   // ConnectionContext
   def https(sslContext: SSLContext,
+            sslConfig: Option[AkkaSSLConfig] = None,
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/scaladsl/ConnectionContext.scala]: this will have to be solved by an overload to not break bin compat

**Comment by ktoso** [comment]
Meh, pushed before MiMa things adjusted, on it.

**Comment by ktoso** [comment]
Validation green so please have a look, would like it merged rather than lingering around :-) @akka/akka-team

**Comment by johanandren** [comment]
LGTM

**Pull request merged**